packages/expo-cli/src/commands/eas-build/utils/expoUpdates.ts

```diff
@@ -0,0 +1,387 @@
+    bundleReactNative.shellScript = `${bundleReactNative.shellScript.replace(
+      /"$/,
+      ''
+    )}${iOSBuildScript}\\n"`;
+  }
+
+  await fs.writeFile(pbxprojPath, project.writeSync());
+
+  const expoPlistPath = getExpoPlistPath(projectDir, pbxprojPath);
+
+  let expoPlist = {};
+
+  if (await fs.pathExists(path.dirname(expoPlistPath))) {
+    const expoPlistContent = await fs.readFile(expoPlistPath, 'utf8');
+    expoPlist = plist.parse(expoPlistContent);
+  }
+
+  const expoPlistContent = plist.build(
+    IOSConfig.Updates.setUpdatesConfig(exp, expoPlist, username, sdkVersion)
+  );
+
+  if (!(await fs.pathExists(path.dirname(expoPlistPath)))) {
+    await fs.mkdirp(path.dirname(expoPlistPath));
+  }
+
+  await fs.writeFile(expoPlistPath, expoPlistContent);
+  await gitAddAsync(expoPlistPath, { intentToAdd: true });
+}
+
+async function isUpdatesConfiguredIOS(
+  projectDir: string,
+  exp: ExpoConfig,
+  username: string | null,
+  sdkVersion: string | null
+) {
+  const pbxprojPath = await getPbxprojPath(projectDir);
+  const project = await getXcodeProject(pbxprojPath);
+  const bundleReactNative = await getBundleReactNativePhase(project);
+
+  if (!bundleReactNative.shellScript.includes(iOSBuildScript)) {
+    return false;
+  }
+
+  const expoPlistPath = getExpoPlistPath(projectDir, pbxprojPath);
+
+  if (!(await fs.pathExists(expoPlistPath))) {
+    return false;
+  }
+
+  const expoPlist = await fs.readFile(expoPlistPath, 'utf8');
+  const expoPlistData = plist.parse(expoPlist);
+
+  const currentSdkVersion = IOSConfig.Updates.getSDKVersion(exp, sdkVersion);
+  const currentRuntimeVersion = IOSConfig.Updates.getRuntimeVersion(exp);
+  const currentUpdateUrl = IOSConfig.Updates.getUpdateUrl(exp, username);
+
+  if (
+    (currentRuntimeVersion
+      ? expoPlistData[IOSConfig.Updates.Config.RUNTIME_VERSION] === currentRuntimeVersion
+      : expoPlistData[IOSConfig.Updates.Config.SDK_VERSION] === currentSdkVersion) &&
+    expoPlistData[IOSConfig.Updates.Config.UPDATE_URL] === currentUpdateUrl
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+async function getPbxprojPath(projectDir: string) {
+  const pbxprojPaths = await new Promise<string[]>((resolve, reject) =>
+    glob('ios/*/project.pbxproj', { absolute: true, cwd: projectDir }, (err, res) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    })
+  );
+
+  const pbxprojPath = pbxprojPaths.length > 0 ? pbxprojPaths[0] : undefined;
+
+  if (!pbxprojPath) {
+    throw new Error("Couldn't find Xcode project");
+  }
+
+  return pbxprojPath;
+}
+
+async function getXcodeProject(pbxprojPath: string) {
+  const project = xcode.project(pbxprojPath);
+
+  await new Promise((resolve, reject) =>
+    project.parse(err => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    })
+  );
+
+  return project;
+}
+
+function getExpoPlistPath(projectDir: string, pbxprojPath: string) {
```

Comment:
similar logic https://github.com/expo/expo-cli/blob/2c51e2f7b549ef6e2b3cb86f721959ad01ae6c5d/packages/expo-cli/src/commands/apply/configureIOSProjectAsync.ts#L86

Reply:
@EvanBacon should I make changes in `@expo/xdl` to be able to use the `IosPlist` utility? We also need a method to read the current plist for checking if it's configured which it doesn't expose.